sources/packages/backend/apps/api/src/route-controllers/student-appeal/_tests_/e2e/student-appeal.students.controller.submitStudentAppeal.e2e-spec.ts

```diff
@@ -0,0 +1,560 @@
+    // Arrange
+    const application = await saveFakeApplication(appDataSource, {
+      student: student,
+    });
+    const endpoint = `/students/appeal/application/${application.id}`;
+    const payload = {
+      studentAppealRequests: [{ formData: {}, formName: "test" }],
+    };
+
+    // Act
+    // Set application status to completed
+    application.applicationStatus = ApplicationStatus.Completed;
+    await applicationRepo.save(application);
+    // Create an existing student appeal for this application
+    const studentAppealRequest = createFakeStudentAppealRequest();
+    const studentAppeal = createFakeStudentAppeal({
+      application,
+      appealRequests: [studentAppealRequest],
+    });
+    await studentAppealRepo.save(studentAppeal);
+    // Set the student appeal status to Pending
+    studentAppealRequest.studentAppeal = studentAppeal;
+    studentAppealRequest.appealStatus = StudentAppealStatus.Pending;
+    await studentAppealRequestRepo.save(studentAppealRequest);
+
+    // Assert
+    await request(app.getHttpServer())
+      .post(endpoint)
+      .auth(studentToken, BEARER_AUTH_TYPE)
+      .send(payload)
+      .expect(HttpStatus.UNPROCESSABLE_ENTITY)
+      .expect({
+        statusCode: HttpStatus.UNPROCESSABLE_ENTITY,
+        message: "There is already a pending appeal for this student.",
+        error: "Unprocessable Entity",
+      });
+
+    // Set the student appeal status to Approved to allow the next test case to be processed
+    studentAppealRequest.appealStatus = StudentAppealStatus.Approved;
+    await studentAppealRequestRepo.save(studentAppealRequest);
+  });
+
+  it("Should throw unprocessable entity exception error when the payload is invalid for formIO dryRun test.", async () => {
+    // Arrange
+    const application = await saveFakeApplication(appDataSource, {
+      student: student,
+    });
+    const endpoint = `/students/appeal/application/${application.id}`;
+    const payload = {
+      studentAppealRequests: [{ formData: {}, formName: "test" }],
+    };
+
+    // Act
+    // Set application status to completed
+    application.applicationStatus = ApplicationStatus.Completed;
+    await applicationRepo.save(application);
+
+    // Mock form.io dry-run submission
+    formService.dryRunSubmission = jest.fn().mockResolvedValue({
+      valid: false,
+      data: { data: payload.studentAppealRequests },
+    });
+
+    // Assert
+    await request(app.getHttpServer())
+      .post(endpoint)
+      .auth(studentToken, BEARER_AUTH_TYPE)
+      .send(payload)
+      .expect(HttpStatus.BAD_REQUEST)
+      .expect({
+        statusCode: HttpStatus.BAD_REQUEST,
+        message: "Not able to submit student appeal due to invalid request.",
+        error: "Bad Request",
+      });
+  });
+
+  it("Should determine student's new income when the student submits an appeal for student new income.", async () => {
```

Comment:
Student income appeal is a POC and I would say we should create E2E for that.

Reply:
@dheepak-aot What is a POC? Where should we create E2E test for this scenario?